src/System.Composition.Convention/src/Microsoft/Composition/Diagnostics/SilverlightTraceWriter.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if !FEATURE_TRACING
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Text;
+
+namespace Microsoft.Composition.Diagnostics
+{
+    internal sealed class DebuggerTraceWriter : TraceWriter
+    {
+        private static readonly string s_sourceName = "System.Composition";
+
+        public override bool CanWriteInformation
+        {
+            get { return false; }
+        }
+
+        public override bool CanWriteWarning
+        {
+            get { return false; /* Debugger.IsLogging(); */ }
+        }
+
+        public override bool CanWriteError
+        {
+            get { return false; /* return Debugger.IsLogging(); */ }
+        }
+
+        public override void WriteInformation(CompositionTraceId traceId, string format, params object[] arguments)
+        {
+            WriteEvent(TraceEventType.Information, traceId, format, arguments);
+        }
+
+        public override void WriteWarning(CompositionTraceId traceId, string format, params object[] arguments)
+        {
+            WriteEvent(TraceEventType.Warning, traceId, format, arguments);
```

Comment:
If CanWriteWarning and CanWriteError are hard coded to return false, should these WriteWarning and WriteError methods instead throw a NotSupportedException?